pkg/logs/input/file/scanner.go

```diff
@@ -185,23 +186,50 @@ func (s *Scanner) launchTailers(source *config.LogSource) {
 		if _, isTailed := s.tailers[file.Path]; isTailed {
 			continue
 		}
-		var tailFromBeginning bool
+
+		mode, found := TailingModeFromString(source.Config.TailingMode)
+		if !found {
+			log.Infof("Invalid or no tailing mode provided, tailing from the end of the file %v", file.Path)
+			source.Config.TailingMode = mode.String()
+		}
+		if wildcard && (mode == Beginning || mode == ForceBeginning) {
```

Comment:
this check could be done when we initialise the source, you probably want to add some logic somewhere in `/pkg/logs/config`, I feel like it's not pure to change the config in the scanner,

Reply:
It sounds reasonable, I will try to relocate this logic.